subsys/usb/usb_c/usbc_pe_common.c

```diff
@@ -720,6 +733,216 @@ void pe_drs_send_swap_exit(void *obj)
+	}
+
+	if (atomic_test_and_clear_bit(pe->flags, PE_FLAGS_SEND_SOFT_RESET)) {
+		/* Send Soft Reset message */
+		pe_send_ctrl_msg(dev, pe->soft_reset_sop, PD_CTRL_SOFT_RESET);
+		return;
+	}
+
+	if (atomic_test_and_clear_bit(pe->flags, PE_FLAGS_MSG_DISCARDED)) {
+		/* Inform Device Policy Manager that the message was discarded */
+		policy_notify(dev, MSG_DISCARDED);
+		pe_set_state(dev, PE_SNK_READY);
+	} else if (atomic_test_and_clear_bit(pe->flags, PE_FLAGS_TX_COMPLETE)) {
+		/* Start SenderResponse timer */
+		usbc_timer_start(&pe->pd_t_sender_response);
+	}
+	/*
+	 * The Policy Engine Shall transition to the PE_SNK_Wait_for_Capabilities
+	 * state when:
+	 *      1: An Accept Message has been received on SOP
+	 */
+	else if (atomic_test_and_clear_bit(pe->flags, PE_FLAGS_MSG_RECEIVED)) {
+		header = prl_rx->emsg.header;
+
+		if (received_control_message(dev, header, PD_CTRL_ACCEPT)) {
+			pe_set_state(dev, PE_SNK_WAIT_FOR_CAPABILITIES);
+		}
+	}
+	/*
+	 * The Policy Engine Shall transition to the PE_SNK_Hard_Reset state when:
+	 *      1: A SenderResponseTimer timeout occurs (Handled in pe_report_error function)
+	 *      2: Or the Protocol Layer indicates that a transmission error has occurred
+	 */
+	else if (usbc_timer_expired(&pe->pd_t_sender_response)) {
+		pe_set_state(dev, PE_SNK_HARD_RESET);
+	}
+}
+
+/**
+ * @brief PE_Send_Soft_Reset Exit State
+ */
+void pe_send_soft_reset_exit(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+
+	/* Stop Sender Response Timer */
+	usbc_timer_stop(&pe->pd_t_sender_response);
+}
+
+
+
+/**
+ * @brief 8.3.3.6.2.1 PE_SNK_Send_Not_Supported State
+ */
+void pe_send_not_supported_entry(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	LOG_INF("PE_Not_Supported");
+
+	/* Request the Protocol Layer to send a Not_Supported or Reject Message. */
+	if (prl_get_rev(dev, PD_PACKET_SOP) > PD_REV20) {
+		pe_send_ctrl_msg(dev, PD_PACKET_SOP, PD_CTRL_NOT_SUPPORTED);
+	} else {
+		pe_send_ctrl_msg(dev, PD_PACKET_SOP, PD_CTRL_REJECT);
+	}
+}
+
+void pe_send_not_supported_run(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+	const struct device *dev = pe->dev;
+
+	if (atomic_test_bit(pe->flags, PE_FLAGS_TX_COMPLETE) ||
+			atomic_test_bit(pe->flags, PE_FLAGS_MSG_DISCARDED)) {
+		atomic_clear_bit(pe->flags, PE_FLAGS_TX_COMPLETE);
+		atomic_clear_bit(pe->flags, PE_FLAGS_MSG_DISCARDED);
+		pe_set_state(dev, PE_SNK_READY);
+	}
+}
+
+/**
+ * @brief 8.3.3.6.2.3 PE_SNK_Chunk_Received State
+ */
+void pe_chunk_received_entry(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+
+	LOG_INF("PE_SNK_Chunk_Received");
+
+	/*
+	 * On entry to the PE_SNK_Chunk_Received state, the Policy Engine
+	 * Shall initialize and run the ChunkingNotSupportedTimer.
+	 */
+	usbc_timer_start(&pe->pd_t_chunking_not_supported);
+}
+
+/**
+ * @brief PE_Chunk_Received Run State
+ */
+void pe_chunk_received_run(void *obj)
```

Comment:
`static void pe_chunk_received_run(void *obj)` ? as it is only used here. Please also check the other removed from [subsys/usb/usb_c/usbc_pe_snk_states_internal.h](https://github.com/zephyrproject-rtos/zephyr/pull/53582/files/1bb44993ce43f25bdd1015edca9ac5b4ec5c8ca9#diff-8cd388c1169e8169381910f8ebec9a9423d7ea6ba1d310330d56d54d3ddb6082)

Reply:
Done